lib/explorer/data_frame.ex

```diff
@@ -104,39 +104,110 @@ defmodule Explorer.DataFrame do
   @enforce_keys [:data, :groups]
   defstruct [:data, :groups]
 
+  @type column :: String.t() | non_neg_integer() | atom()
+  @type columns :: [column] | Range.t()
+  @type columns_pairs(other) :: [{column(), other}] | %{column() => other}
+
   @default_infer_schema_length 1000
 
+  # Guards and helpers for columns
+
+  defguard is_column(column) when is_binary(column) or is_atom(column) or is_integer(column)
+  defguard is_column_pairs(columns) when is_list(columns) or is_map(columns)
+
+  defp to_column(_, column) when is_atom(column), do: Atom.to_string(column)
+  defp to_column(_, column) when is_binary(column), do: column
+  defp to_column(df, column) when is_integer(column), do: to_column!(df, column)
+
+  defp to_column!(df, column) when is_atom(column), do: to_column!(df, Atom.to_string(column))
```

Comment:
From an idiomatic standpoint, I'm not understanding the reasoning for bang and non-bang versions of this here. Asking so I can learn :).

Reply:
I should have added a comment about that :grimacing: 
The difference is that the bang version will raise an exception if the column is not present. But the problem here is that the version without a bang that reads an integer column also raises. I left this way because I can´t guarantee that an integer column will ever be valid to be converted. Not sure if this is going to be confusing :/

Reply:
Makes sense to me! 🙏